dbschema/project.esdl

```diff
@@ -153,7 +152,107 @@ module default {
           period := reportRange
         })
       )
-    )
+    );
+
+    trigger addRemovePeriodicReports after update for each 
+      when (
+        __old__.mouStart != __new__.mouStart
+        or __old__.mouEnd != __new__.mouEnd
+        or __old__.financialReportPeriod != __new__.financialReportPeriod
+      ) 
+      do (
+      with
+        existingReportPeriods := (
+            select FinancialReport
+            filter .container.id = __old__.id
+        ).period,
+        interval := (
+          select (if __new__.financialReportPeriod = default::ReportPeriod.Monthly then '1' else '3')
+          ),
+        requestedReportPeriods := Project::create_periodic_report_ranges(
+          __new__.mouStart,
+          __new__.mouEnd,
+          interval
+        )
+      if __old__.financialReportPeriod != __new__.financialReportPeriod (
```

Comment:
@CarsonF  - The code within these conditional blocks works; I've tested it just like I have the other trigger above.  However, I'm having some issue with the syntax on this `if` statement here.  I'm getting a compilation error on this line that says:

> error: Missing '['

Just before the parenthesis.  GPT and the documentation have been most unhelpful in explaining why.

Reply:
Missing `then` between condition and truth expression

Reply:
Well, no.  That's how I got here.  With that in place we get this:

> error: Unexpected keyword 'THEN'

Apparently within triggers only you don't use the keyword `then`.  But...that also isn't working as you saw above.

Reply:
![Named-Only_Arguments_-_User-Defined_Functions___Tutorial___EdgeDB_Docs](https://github.com/user-attachments/assets/b4e06e6a-7990-46a7-b9c8-669fde3df6af)

Reply:
I can't get either way to work, tbh.

Reply:
The AI is lying to you. It does that a lot with EdgeDB. I never trust its answers either way in this context unfortunately.

Reply:
Oh you have a `with` expression preceding this. You need to wrap the `if` in a `select`

Reply:
Ah!  The `select` wrap was it!  I thought I tried that, but apparently not.  🤷🏼 

And yes, I have had some less than stellar answers with the AI as well.  But since I couldn't get it working either way I didn't know who was right.